lib/acts_as_mentionable/mentioner_parser.rb

```diff
@@ -0,0 +1,24 @@
+module ActsAsMentionable
+  class MentionerParser
+    attr_reader :mentioner
+
+    def initialize mentioner
+      @mentioner = mentioner
+    end
+
+    def parse!
+      mentionable_templates = mentioner.send(mentioner.class.mention_parsed_field).scan(template_for_parsing)
+      # => ["{@|9999|user}", "{@|123|user}"]
+      mentionable_templates.map do |template|
+        _delimiter, mentionable_id, mentionable_klass = template.gsub(/[\{,\}]/, '').split('|')
+        mentionable_klass.classify.constantize.find mentionable_id
+      end
+    end
+
+    private
+
+      def template_for_parsing
+        /\{\W\|\d+\|\w+\}/
```

Comment:
Are we going to use a different indicator symbols except `@`? Because in this regex we are assuming that it can be any non-word character and we don't care which one is it.
Also, we can use grouping in regexp, so we don't need to parse it again above in `template.gsub(/[\{,\}]/, '').split('|')`
```suggestion
        /\{@\|(\d+)\|(\w+)\}/
```

```ruby
irb(main):025:0> 'Lorem {@|42|user} ipsum {@|256|community} dolor'.scan(/\{@\|(\d+)\|(\w+)\}/)
=> [["42", "user"], ["256", "community"]]

[["42", "user"], ["256", "community"]].map do |mentionable_id, mentionable_type|
  # ...
end
```

Reply:
But you can't gsub the ["256", "community"] array in string. That's why I gsub this template, and get  "{@|123|user}" inclusions to replace it with @ user .

Reply:
I think `#scan` is also available:
```ruby
template = "Hi, {@|123|user}! Welcome to {@|123|community}"
mention_matching_regex = /\{@\|(\d+)\|(\w+)\}/
template.scan(mention_matching_regex)
# => [["123", "user"], ["123", "community"]]
```

Reply:
Yeah, but if now I want to replace the [{@|123|user}, {@|123|community}] in original template: With *san* I should scan template again to find {@|123|user} and replace it with "@user123name" . 
In my code I do it in one iteration: gsub with block and while i've found the inclusion, I processing the mentionable and return the correct template for replacement in the end of the block. As result i've got the array of mentionables in mentionables var and as result of gsub block the body like 
"Hi, @123username Welcome to @123community"